src/ui/input/color-picker/hidden-color-input.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { PureComponent, ChangeEvent } from 'react';
+import { Input } from '@teambit/base-ui.input.checkbox.label';
+
+export type HiddenColorInputProps = React.InputHTMLAttributes<HTMLInputElement>;
+
+type HiddenColorInputState = {
+  isOpen: boolean;
+  type: string;
+};
+
+export class HiddenColorInput extends PureComponent<HiddenColorInputProps, HiddenColorInputState> {
+  state = {
+    isOpen: false,
+    type: 'color',
+  };
+
+  handleOnChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = this.props;
+    onChange && onChange(e);
+  };
+
+  handleOnClick = (e: any) => {
+    const { isOpen } = this.state;
+    if (isOpen) {
+      this.setState({ isOpen: false, type: 'text' });
+      setTimeout(() => this.setState({ type: 'color' }), 500);
```

Comment:
500ms is too long, try setTimeout without a timer, or with 1ms 10ms.

Reply:
also set to `props.type`

Reply:
this doesn't cover all cases, for example, if clicking outside.
I think `onChange` will work better here, without the `isOpen` state

Reply:
You right, I need to handle this use case also.

Reply:
done